File: Webclient/src/app/module/armory/domain_value/character_history.ts
import {ArenaTeam} from "./arena_team";
import {CharacterFacial} from "./character_facial";
import {CharacterGuild} from "./character_guild";
import {CharacterInfo} from "./character_info";

export interface CharacterHistory {
    id: number;
    character_id: number;
    character_info: CharacterInfo;
    character_name: string;
    character_guild: CharacterGuild | null;
    character_title: number | null;
    profession_skill_points1: number | null;
    profession_skill_points2: number | null;
    facial: CharacterFacial | null;
    arena_teams: Array<ArenaTeam>;
    timestamp: number;
}

File: Exporter/Backend/src/modules/transport_layer/domain_value/character_history.rs
use crate::modules::transport_layer::{ArenaTeam, CharacterFacialDto, CharacterGuildDto, CharacterInfoDto};

#[derive(Debug, Clone, Serialize)]
pub struct CharacterHistoryDto {
    pub character_info: CharacterInfoDto,
    pub character_name: String,
    pub character_guild: Option<CharacterGuildDto>,
    pub character_title: Option<u16>,
    pub profession_skill_points1: Option<u16>,
    pub profession_skill_points2: Option<u16>,
    pub facial: Option<CharacterFacialDto>,
    pub arena_teams: Vec<ArenaTeam>,
}
